### DISCUSSION TITLE
NEXTAUTH OIDC CONNECT Configuration

### DISCUSSION THREAD

**Discussion by Lefort-Antoine** [lobehub/lobehub#4398] [Self Hosting] [answered]
[docker-compose.txt](https://github.com/user-attachments/files/17415008/docker-compose.txt)

Hi,
I'm trying to configure following this documentation 
https://lobehub.com/fr/docs/self-hosting/server-database/docker-compose

I have a OIDC with this .env configue:

`

OPENAI_API_KEY=sk-XXXXX
OPENAI_PROXY_URL=https://api.openai.com/v1

LOBE_PORT=3210
MINIO_PORT=9000

LOBE_DB_NAME=lobechat
POSTGRES_PASSWORD=uWNZugjBqixf8dxC

NEXTAUTH_SECRET=NX2kaPE923dt6BL2U8e9oSre5RfoT7hg

AUTH_OIDC_CLIENT_ID=XXX
AUTH_OIDC_CLIENT_SECRET=XXXX
AUTH_OIDC_ISSUER=https://XXX.fr/

OAUTH2_PROXY_EMAIL_DOMAINS=XX.fr
OAUTH2_PROXY_COOKIE_SECRET=XXX
OAUTH2_PROXY_CLIENT_ID=XXX
OAUTH2_PROXY_CLIENT_SECRET=XXX
OAUTH2_PROXY_SCOPE=openid 
OAUTH2_PROXY_PROVIDER=oidc
OAUTH2_PROXY_REDIRECT_URL=http://X.X.X.X:3210/oauth2/callback
OAUTH2_PROXY_OIDC_ISSUER_URL=https://XXXX.fr/
OAUTH2_PORT=3000
`

This is my error in logs: 
`
next auth: undefined
lobe-database  | next auth: undefined
lobe-database  | {"level":30,"time":1729178270345,"pid":29,"hostname":"f644ab03ac5a","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
lobe-database  | a [TRPCError]: UNAUTHORIZED
lobe-database  |     at /app/.next/server/chunks/39184.js:1:1899
lobe-database  |     at c (/app/.next/server/chunks/30828.js:4:74)
lobe-database  |     at r (/app/.next/server/chunks/30828.js:1:5657)
lobe-database  |     at /app/.next/server/chunks/52145.js:1:7166
lobe-database  |     at Array.map (<anonymous>)
lobe-database  |     at y (/app/.next/server/chunks/52145.js:1:6727)
lobe-database  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
lobe-database  |     at async n (/app/.next/server/app/(backend)/trpc/lambda/[trpc]/route.js:1:3750)
lobe-database  |     at async /app/node_modules/.pnpm/next@14.2.8_@babel+core@7.23.6_supports-color@8.1.1__@opentelemetry+api@1.9.0_react-dom@18.3._itjmu72s7n7tov6po6nkhcdpya/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36932
`

Thanks for help

**Comment by arvinxx** [comment]
I think we don't support OAUTH2 now.  @cy948 Can you check it?

**Comment by cy948** [comment, marked as answer]
Only OIDC protocol supported.
